### PR TITLE
rehydration added

### PR DESCRIPTION
**Pull request by bishwobhandari**

### CODE REVIEW

File: frontend/src/stores/DataStore.js [hunk 1]
```diff
@@ -1,25 +1,297 @@
...
+
+    @action saveOptions = (name, value, id) => {
+        this.usersInput.options[id][name] = value;
+    };
+
+    @action deleteOptions = val => {
+        this.usersInput.options.splice(val, 1);
+    };
+
+    // @action showDataForm = formType => {
```
Comment (shapiromatron) [inline frontend/src/stores/DataStore.js:24]: remove?

File: frontend/src/stores/DataStore.js [hunk 2]
```diff
@@ -1,25 +1,297 @@
...
+                    !this.prior_weight_models.includes(val.name)
+                ) {
+                    val.prior_weight = 0;
+                }
+            });
+        });
+    };
+
+    //returns the dataset which are enabled
+    @action getEnabledDataset() {
```
Comment (shapiromatron) [inline frontend/src/stores/DataStore.js:147]: this should be a `@computed`

File: frontend/src/stores/DataStore.js [hunk 3]
```diff
@@ -1,25 +1,297 @@
...
+        let obj = toJS(this.savedDataset).filter(item => item.enabled == true);
+        return obj;
+    }
+
+    //returns enabled model types
+    @action getModels() {
```
Comment (shapiromatron) [inline frontend/src/stores/DataStore.js:153]: should this also be computed? as i understand it, actions should only modify content in `observables`

File: frontend/src/components/Main/DatasetName.js [hunk 4]
```diff
@@ -12,47 +12,58 @@ class DatasetName extends Component {
...
+                                                {this.props.DataStore.AdverseDirectionList.map(
+                                                    (item, i) => {
+                                                        return [
+                                                            <option key={i} value={item.value}>
+                                                                {item.name}
+                                                            </option>,
+                                                        ];
+                                                    }
+                                                )}
+                                                {/* <option value="automatic">automatic</option>
```
Comment (shapiromatron) [inline frontend/src/components/Main/DatasetName.js:59]: delete this text

File: frontend/src/stores/DataStore.js (hunk 3 above)
Comment (bishwobhandari) [reply frontend/src/stores/DataStore.js]: changed to @computed

File: frontend/src/components/Main/DatasetName.js (hunk 4 above)
Comment (bishwobhandari) [reply frontend/src/components/Main/DatasetName.js]: deleted

File: frontend/src/stores/DataStore.js (hunk 2 above)
Comment (bishwobhandari) [reply frontend/src/stores/DataStore.js]: changed to @computed

**Pull request merged**